app/models/manageiq/providers/ibm_power_hmc/inventory/collector/infra_manager.rb

```diff
@@ -1,68 +1,13 @@
 class ManageIQ::Providers::IbmPowerHmc::Inventory::Collector::InfraManager < ManageIQ::Providers::IbmPowerHmc::Inventory::Collector
-  def initialize(manager, target)
-    super
-    @netadapters = {}
-    @sriov_elps = {}
-    @vnics = {}
-    @ssps = {}
-    @pcm_enabled = {}
-  end
-
   def collect!
```

Comment:
I didn't realize you were calling `collect!` manually in the parser [[before]](https://github.com/ManageIQ/manageiq-providers-ibm_power_hmc/pull/88/files#diff-9f4a5a239e01b764383415b3f1422553029e542f8ab3e0b7c53b7c05500e31a3L4)

If you want to pre-collect inventory (typically for performance reasons) you should define a `collect` method that will be called automatically prior to parsing (https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/base_manager/refresher.rb#L118 which calls https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/inventory.rb#L32)

Reply:
Most providers don't define this and lazy fetch inventory when the parser calls it though which is what you're doing here so I think you can just delete this method entirely.

Reply:
Unfortunately, we still need to call it from the parser because we want to close the HMC session at the end.